src/main/scala/edu/ie3/simona/agent/grid/PowerFlowSupport.scala

```diff
@@ -77,93 +82,195 @@
       gridMainRefSystem: RefSystem,
       targetVoltageFromReceivedData: Boolean = true,
       ignoreTargetVoltage: Boolean = false
-  ): Array[PresetData] = {
-
-    val mainRefSystemPowerUnit = gridMainRefSystem.nominalPower.getUnit
+  ): (Array[PresetData], WithForcedStartVoltages) =
+    nodes.foldLeft((Array.empty[PresetData], Array.empty[StateData])) {
```

Comment:
One could think about replacing this by a combination of `map`, `unzip` and `flatten`, which I believe would be a bit cleaner